### PR TITLE
OWLS-66101, Site documentation for persistent volumes

### PR DESCRIPTION
**Pull request by jgrundback**

### CODE REVIEW

**Comment by jgrundback** [comment]
Comment from Tom B:

My first cut comment:   (1) I don’t think we’re restricting the PV type to NFS & HOST_PATH  (2) the k8s object names in the readme are arbitrary (suggested) not required, so maybe that should be made clear…

**Comment by tbarnes-us** [comment]
One more comment, if a PV/PVC is associated with a particular domain-uid, it’s a best practice to label it with weblogic.domainUID=mydomainuid — this makes it easy to search for the resources associated with a particular domain, it also helps the domain cleanup sample script (which uses labels).

**Comment by jgrundback** [comment]
@doxiao Yes, in fact it is based on that readme. I guess this is meant as a general PV documentation whereas the sample one is more specific.

File: site/persistent-volumes.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+## Prerequisites
+
+The following prerequisites must be fulfilled before proceeding with the creation of the persistent volume.
+* Create a Kubernetes namespace for the persistent volume claim unless the intention is to use the default namespace. Note that a PVC has to be in the same namespace as the domain resource that uses it.
+* Make sure that the host directory that will be used as the persistent volume already exists and has the appropriate file permissions set.
+
+
+# YAML templates
+
+Persistent volumes and claims and their properties are described in YAML files. For each PV, you should create one PV YAML and one PVC YAML file. Below you can find two YAML tempaltes, one for the PV and one for the PVC. Each template references the following configuration parameters.
```
Comment (rosemarymarano) [inline site/persistent-volumes.md]: create one PV YAML and -> create one PV YAML file and

Comment (rosemarymarano) [inline site/persistent-volumes.md]: Below you can find two YAML tempaltes (typo)->Below you can find two YAML templates

**Comment by doxiao** [comment]
Two docs are mostly duplicate. If we want to keep both, we need to simplify one of them and pointing to the other.

**Comment by doxiao** [comment]
I am wondering if this works.

The main site doc does not provide any samples/templates, instead it just introduces the PV ad PVC, where they can be used with the Operator - say domain home on pv use case or log persistence, dedicated PV vs shared PVC, as well as the common problem with PV.  The sample readme provides the details with samples. The main site doc points to the sample readme.

**Comment by mriccell** [comment]
I agree with Dongbo that the main documentation should describe PV/PVC and the use in this project to persist optionally the domain home, the server logs, and the application binaries. It should describe that if thedomain home is persisted to the PV/PVC then it must be on a shared storage across all the servers in the WebLogic domain.  This main document should describe any issues that might be encountered with PV/PVCs (they are described in the blog https://blogs.oracle.com/weblogicserver/how-to-run-weblogic-clusters-on-the-oracle-cloud-infrastructure-container-engine-for-kubernetes).  This document should link to the sample which describes how to create a PV/PVC in detail.

**Comment by jgrundback** [comment]
@tbarnes-us @rosemarymarano I think I resolved your comments, thank you!

**Comment by jgrundback** [comment]
@mriccell @doxiao I am looking into combining this with the samples.

**Comment by doxiao** [comment]
Can we keep the configuration, templates, examples and checking of the results in the sample readme  as we agreed upon?

**Comment by jgrundback** [comment]
@doxiao Thank you! Is that better now?

**Comment by jgrundback** [comment]
@doxiao Thank you again. I took out the naming convention section, but kept the generic parts regarding YAML files, plus I rewrote the dedicated domain part.

**Comment by jgrundback** [comment]
@doxiao @mriccell I kept the generic kubectl commands used to create and query kubernetes resources. I do think they belong in the general doc as they are important to the process of creating persistent volumes.

**Comment by jgrundback** [comment]
@doxiao @mriccell @rosemarymarano Hi again, sorry to bug everyone. I would like to close this one out, so can we go over what is still missing?

**Comment by rjeberhard** [comment]
@rosemarymarano @mriccell @jgrundback Cycling back to this one, as well.  Are we ready to commit this or is there more work?

**Comment by rosemarymarano** [comment]
I think it looks OK. I'll have a better opportunity to compare it to the PV doc in the samples after it is committed.

**Comment by jgrundback** [comment]
@mriccell @doxiao I added what I think we need from the review earlier this week.

**Comment by jgrundback** [comment]
@rosemarymarano You can review the rendered version here:
https://github.com/oracle/weblogic-kubernetes-operator/blob/feature/OWLS-66101/site/persistent-volumes.md

**Comment by rosemarymarano** [comment]
It looks fine to me.

**Pull request merged**